### PR TITLE
Fixes security camera range issues v2

### LINKED ISSUE #63607: Security Cam Inconsistency
## Round ID:
Current

## Testmerges:
Current

## Reproduction:
Adjacency checks for security cams (introduced by me in #63451) cause borgs to be unable to view cams from a distance.
It also does not apply to secureye, the tablet program.
Thx @SpookyTheFox for this bug report.

### PR DESCRIPTION
**Pull request by jlsnow301**
## About The Pull Request
A little history on this issue...

Security cameras were viewable from a huge distance, thus an issue spawned over it. Original range:

The source of the issue is this: TGUIs have a default range. To balance this, they do not update from a certain distance (red eye).
With security cams, this is not valid, even at max range the camera overlay updates.
(if you're counting the camera list, that doesn't update, but that's not the point here)

I implemented an adjacency check, which is more of a balance change. It wasn't satisfactory, and I thought about reverting it.
After talking with @LemonInTheDark, I agreed with their idea of making a cutoff for UI_DISABLED.
the quote
Screenshot 2021-12-29 231315

## Why It's Good For The Game
Fixes #63607
A little more consistent and logical camera usage

## Changelog

:cl:
fix: Cameras are once again viewable from a distance, but this is limited to 2 tiles.
fix: The viewing range limit of 2 tiles also applies to the SecurEye tablet software.
/:cl:

### CODE REVIEW

File: code/modules/tgui/states/cameras.dm [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * update. That is, you can watch cameras at huge ranges
+ * regardless. camera_state disables long range viewing
+ * and observers (they can just go there!) but leaves it
+ * interactive for silicons.
+ *
+ */
+
+GLOBAL_DATUM_INIT(camera_state, /datum/ui_state/camera_state, new)
+
+/datum/ui_state/camera_state/can_use_topic(src_object, mob/user)
```
Comment (Mothblocks) [inline code/modules/tgui/states/cameras.dm:15]: Don't make a new `ui_state`, they're old NanoUI junk.

Override `ui_status` instead.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/modules/tgui/states/cameras.dm (hunk 1 above)
Comment (jlsnow301) [reply code/modules/tgui/states/cameras.dm:15]: oh! i thought it was the inverse. will do

Comment (LemonInTheDark) [reply code/modules/tgui/states/cameras.dm:15]: WTF none ever told me that.

**Comment by Mothblocks** [comment]
Remove the semicolons and it's good

**Comment by LemonInTheDark** [comment]
It'd be nice if the distance wasn't hardcoded, since really we just want to act like the standard ui state, but we close if we're supposed to not update

**Comment by jlsnow301** [comment]
> It'd be nice if the distance wasn't hardcoded, since really we just want to act like the standard ui state, but we close if we're supposed to not update

i can do this but note that it currently gives it a 3 tile range. if i set it to close on disabled, you'll have only 2

File: code/game/machinery/computer/camera.dm [hunk 2]
```diff
@@ -91,7 +91,8 @@
 		ui.open()
 
 /obj/machinery/computer/security/ui_status(mob/user)
-	if(!in_range(user, src) && !isobserver(user))
+	. = ..()
+	if(!.)
```
Comment (LemonInTheDark) [inline code/game/machinery/computer/camera.dm:95]: I would prefer a direct check for `UI_DISABLED` here

**Pull request merged**